extensions/cornerstone/src/init.js

```diff
@@ -107,6 +109,98 @@ export default function init({ servicesManager, configuration }) {
     tools.push(...toolsGroupedByType[toolsGroup])
   );
 
+  /* MeasurementService configuration */
+  const csToolsVer4MeasurementSource = MeasurementService.createSource('CornerstoneTools', '4');
+  const { addOrUpdate, getAnnotation } = csToolsVer4MeasurementSource;
+
+  /* Add mapping criterias to measurement service */
+  const matchingCriteria = {
+    valueType: MeasurementService.VALUE_TYPES.POLYLINE,
+    points: 2,
+  };
+  MeasurementService.addMapping(
+    csToolsVer4MeasurementSource,
+    'Length',
+    matchingCriteria,
+    toAnnotation,
+    toMeasurement
+  );
+
+  /* Measurement Service Events */
+  cornerstone.events.addEventListener(
+    cornerstone.EVENTS.ELEMENT_ENABLED,
+    event => {
+      const {
+        MEASUREMENT_ADDED,
+        MEASUREMENT_UPDATED,
+      } = MeasurementService.EVENTS;
+
+      MeasurementService.subscribe(
+        MEASUREMENT_ADDED,
+        ({ source, measurement }) => {
+          if (![csToolsVer4MeasurementSource.id].includes(source.id)) {
+            const annotation = getAnnotation('Length', measurement.id);
+
+            console.log(
+              '[subscriber::MEASUREMENT_ADDED] Measurement added',
+              measurement,
+            );
+            console.log('Mapped annotation to be saved:', annotation);
+          }
+        });
+
+      MeasurementService.subscribe(
+        MEASUREMENT_UPDATED,
+        ({ source, measurement }) => {
+          if (![csToolsVer4MeasurementSource.id].includes(source.id)) {
+            const annotation = getAnnotation('Length', measurement.id);
+
+            console.log(
+              '[subscriber::MEASUREMENT_UPDATED] Measurement updated',
+              measurement
+            );
+            console.log('Mapped annotation to be saved:', annotation);
+          }
+        }
+      );
+
+      const addOrUpdateMeasurement = eventData => {
+        try {
+          const measurementServiceId = addOrUpdate('Length', eventData);
+
+          if (!eventData.measurementData._measurementServiceId) {
+            addMeasurementServiceId(measurementServiceId, eventData);
+          }
+        } catch (error) {
+          console.warn('Failed to add or update measurement in measurement service:', error);
+        }
+      };
+
+      const addMeasurementServiceId = (id, eventData) => {
+        const { measurementData } = eventData;
+        Object.assign(measurementData, { _measurementServiceId: id });
+      };
+
+      event.detail.element.addEventListener(
+        csTools.EVENTS.MEASUREMENT_MODIFIED,
+        event => {
+          console.log('[MEASUREMENT_MODIFIED] Updating measurement...', event.detail);
+          addOrUpdateMeasurement(event.detail);
```

Comment:
If `event.detail` is the CornerstoneAnnotation, we could alias to make that more clear

Reply:
If the listeners for these are similar, we could do:

```js
[csTools.events.MEASUREMENT_ADDED, csTools.events.MEASUREMENT_MODIFIED].forEach(csToolsEvtName => {
  event.detail.element.addEventListener(csToolsEvtName, { detail: csToolsAnnotation } => addOrUpdateMeasurement(csToolsAnnotation)); 
})
```

Reply:
Updated, definitely more descriptive, thanks.